tests/dist/mpi/test_remote_execution.cpp

```diff
@@ -0,0 +1,34 @@
+#include <catch2/catch.hpp>
+
+#include "fixtures.h"
+
+#include <faabric/scheduler/Scheduler.h>
+
+namespace tests {
+
+TEST_CASE_METHOD(DistTestsFixture,
+                 "Test running an MPI function spanning two hosts",
+                 "[mpi]")
+{
+    // Set up this host's resources
+    int nLocalSlots = 2;
+    faabric::HostResources res;
+    res.set_slots(nLocalSlots);
+    sch.setThisHostResources(res);
+
+    // Set up the message
+    std::shared_ptr<faabric::BatchExecuteRequest> req =
+      faabric::util::batchExecFactory("mpi", "mpi_bcast", 1);
+    faabric::Message& msg = req->mutable_messages()->at(0);
+    msg.set_ismpi(true);
+    msg.set_mpiworldsize(4);
+
+    // Call the functions
+    sch.callFunctions(req);
```

Comment:
It would be nice to somehow test that these had been executed on different hosts. Could we perhaps check the execution graph? Would be a nice indirect test of that working in a distributed environment too.

Reply:
Yes, I think this is a very valid point. I add a utility function in faasm/faabric#178 and then check for it here.